diff -urpN version.dist version.dist
--- version.dist	2014-09-02 06:23:18.000000000 -0400
+++ version.dist	1969-12-31 19:00:00.000000000 -0500
@@ -1,13 +0,0 @@
-/*
- * version.h: A plugin/program for the Video Disk Recorder
- *
- * See the README file for copyright information and how to reach the author.
- *
- */
-
-#ifndef __version_h_
-#define __version_h_
-
-static const char *VERSION        = "0.1.5pre";
-
-#endif
diff -urpN version.h version.h
--- version.h	1969-12-31 19:00:00.000000000 -0500
+++ version.h	2014-09-02 06:23:18.000000000 -0400
@@ -0,0 +1,13 @@
+/*
+ * version.h: A plugin/program for the Video Disk Recorder
+ *
+ * See the README file for copyright information and how to reach the author.
+ *
+ */
+
+#ifndef __version_h_
+#define __version_h_
+
+static const char *VERSION        = "0.1.5pre";
+
+#endif
