diff -uprN Makefile Makefile
--- Makefile	2014-08-09 17:07:51.000000000 -0400
+++ Makefile	2014-08-19 11:10:47.000000000 -0400
@@ -20,6 +20,7 @@ PKGCFG = $(if $(VDRDIR),$(shell pkg-conf
 LIBDIR = $(call PKGCFG,libdir)
 LOCDIR = $(call PKGCFG,locdir)
 PLGCFG = $(call PKGCFG,plgcfg)
+CONFDIR = $(call PKGCFG,configdir)
 #
 TMPDIR ?= /tmp
 
@@ -37,8 +38,8 @@ APIVERSION = $(call PKGCFG,apiversion)
 -include $(PLGCFG)
 
 ### export all vars for sub-makes, using absolute paths
-LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
-LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
+#LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
+#LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
 export
 unexport PLUGIN
 
@@ -87,6 +88,8 @@ install-server: server
 	# installs to $(LIBDIR)/libvdr-streamdev-server.so.$(APIVERSION)
 
 install: install-client install-server
+	@mkdir -p $(DESTDIR)$(CONFDIR)/plugins/
+	@cp -ar streamdev-server $(DESTDIR)$(CONFDIR)/plugins
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
